src/main/java/org/opensearch/security/transport/SecurityInterceptor.java

```diff
@@ -129,6 +129,15 @@ public <T extends TransportRequest> SecurityRequestHandler<T> getHandler(String
         );
     }
 
+    private User determineUser(Connection connection) {
+        User user0 = getThreadContext().getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+        // pluginUser did not exist prior to 2.19.0
+        if (user0 != null && user0.isPluginUser() && connection.getVersion().before(Version.V_2_19_0)) {
```

Comment:
The logic here is a little confusing to me: since plugin users did not exist before 2.19, when would this be true?